stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/adapter/view/internal/FileAttachmentsView.kt

```diff
@@ -230,31 +226,18 @@ private class FileAttachmentViewHolder(
 
     private fun handleInProgressAttachment(fileSizeView: TextView) {
         attachment?.let { attachment ->
-            attachment.uploadId?.let(ProgressTrackerFactory::getOrCreate)?.let { tracker ->
-                val progress = tracker.currentProgress()
-                val completion = tracker.isComplete()
-                val totalValue = MediaStringUtil.convertFileSizeByteCount(attachment.upload?.length() ?: 0)
-
-                val fileProgress = progress.combine(completion, ::Pair)
-
-                clearScope()
-                scope = CoroutineScope(DispatcherProvider.Main)
-
-                scope?.launch {
-                    fileProgress.collect { (progress, isComplete) ->
-                        val uploadedBytes = (progress / 100F * tracker.maxValue).toLong()
-                        updateProgress(
-                            context,
-                            fileSizeView,
-                            binding.progressBar,
-                            attachment,
-                            uploadedBytes,
-                            isComplete,
-                            totalValue
-                        )
-                    }
-                }
-            }
+            val uploadState = (attachment.uploadState as Attachment.UploadState.InProgress)
```

Comment:
We could take `uploadState` as a parameter to this method to avoid this downcasting

Reply:
Not sure to understand this.

Reply:
We could add a second parameter to the `handleInProgressAttachment`, which has the `InProgress` type, and we could pass in that value at the call site. That way we wouldn't have to do this downcast (from the `UploadState` type to the more specific `InProgress` type) which could be unsafe if we call this method from an incorrect place in the code.